xc/device.ml

```diff
@@ -1912,6 +1912,30 @@ let stop ~xs ~qemu_domid domid  =
     stop_vgpu ();
     stop_qemu ()
 
+let is_hvm_linux domid =
```

Comment:
I think the code is fine but I am a bit puzzled about the use of options. Would it not be enough for `is_hvm_linux` to return a boolean? The use of a `bool option` does not seem to add any value.